cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -120,16 +120,20 @@ macro (build_addon target prefix libs)
             foreach(src_file ${USED_SOURCES})
               file(STRINGS ${src_file} BIN_ADDON_SRC_PARTS)
               foreach(loop_var ${BIN_ADDON_SRC_PARTS})
+                string(REGEX REPLACE ".*#include" "#include" loop_var "${loop_var}")
                 string(FIND "${loop_var}" "#include" matchres)
                 if("${matchres}" EQUAL 0)
                   string(REPLACE " " ";" loop_var "${loop_var}")
                   list(GET loop_var 1 include_name)
                   string(REGEX REPLACE "[<>\"]|kodi/" "" include_name "${include_name}")
                   if(include_name MATCHES ${depend_header})
-                    set(ADDON_DEPENDS "${ADDON_DEPENDS}\n<import addon=\"${${xml_entry_name}}\" version=\"${${depends_name}}\"/>")
-                    # Inform with them the addon header about used type
-                    add_definitions(-D${used_type_name})
-                    message(STATUS "Added usage definition: ${used_type_name}")
+                    get_directory_property(CURRENT_DEFS COMPILE_DEFINITIONS)
+                    if(NOT CURRENT_DEFS MATCHES "${used_type_name}")
```

Comment:
`MATCHES` is wrong as it would also match definitions which contain `${used_type_name}`, but have additional characters at the beginning or end.

Reply:
Not know direct a good alternative.
e.g. on Linux as definition it have `BUILD_KODI_ADDON;HAS_GL=1;ADDON_GLOBAL_VERSION_MAIN_USED;ADDON_GLOBAL_VERSION_GUI_USED;ADDON_INSTANCE_VERSION_SCREENSAVER_USED`, but not sure if this is on all OS like them.

But do not look directly dangerous as it always has this at the beginning and end:
- `ADDON_*`
- `*_USED`

Reply:
```suggestion
                    if(NOT used_type_name IN_LIST CURRENT_DEFS)
```

Reply:
Thanks a lot! This `IN_LIST` is new for me, but it works good here.

With the "not sure" I meant before that mostly in `add_definitions()` a `-D ...` is to begin, and thought that this may not always be a list, but actually only makes sense as a list.